src/h5web/toolbar/controls/DomainSlider/ColorBarGradient.tsx

```diff
@@ -0,0 +1,38 @@
+import { range } from 'lodash';
+import type { SVGProps } from 'react';
+import type { ColorMap } from '../../../../packages/lib';
+import { getInterpolator } from '../../../vis-packs/core/heatmap/utils';
+
+const GRADIENT_PRECISION = 1 / 20;
+const GRADIENT_RANGE = range(0, 1 + GRADIENT_PRECISION, GRADIENT_PRECISION);
+
+interface Props extends SVGProps<SVGRectElement> {
+  colorMap: ColorMap;
+  invertColorMap?: boolean;
+}
+
+function ColorBarGradient(props: Props) {
+  const { colorMap, invertColorMap, ...rectProps } = props;
+  const interpolator = getInterpolator(colorMap, !!invertColorMap);
+
+  return (
+    <>
+      <defs>
+        <linearGradient id="ColorMapGradient" x1="0" x2="1" y1="0" y2="0">
+          {GRADIENT_RANGE.map((v) => {
+            return (
+              <stop
+                key={v}
+                offset={`${v * 100}%`}
+                stopColor={interpolator(v)}
+              />
+            );
+          })}
+        </linearGradient>
+      </defs>
+      <rect {...rectProps} fill="url(#ColorMapGradient)" />
```

Comment:
I could restrict `rectProps` to `x,y,height,width` if you think that's better

Reply:
Yeah it will be stricter, and with `Required<Pick<SVGProps<SVGRectElement>>, 'x' | ...>>` so they are required too. Unless those four props are already required in `SVGProps<SVGRectElement>`? If so, then don't bother.